Library/Homebrew/cmd/desc.rb

```diff
@@ -0,0 +1,47 @@
+require 'descriptions'
+require 'cmd/search'
+
+module Homebrew
+  def desc
+    options = ARGV.options_only
+    cache_op = false
```

Comment:
This `cache_op` along with below logics is confusing. I suggest you use `if-else` flaw with different methods like this:

```ruby
def desc
  if ARGV.include? "--delete-cache"
    Descriptions.delete_cache
  elsif ARGV.include? "--create-cache"
    Descriptions.generate_cache
  elsif ARGV.options_only.empty?
    results = Descriptions.named(ARGV.formulae.map(&:full_name)
    Descriptions.print(results)
  else
    ...search...
  end
end
```